docs/getting_started/flows.md

```diff
@@ -31,6 +31,24 @@ In Windmill, both Scripts and Flows consist of two parts:
 - **Logic**, which in case of a Script is the code, and in case of a Flow 
   is the list of Scripts to be executed.
 
+### For-loops
+
+One functionality that is specific to Flows is the ability to wrap a part 
+of its sequence into a loop.
+
+The expected input is a list, and it is usually coming from either previous
+steps (see this [example][hub-flow2]), or from the Flow's inputs. 
+The variable name to iterate over, can be defined in the `Iterator 
+expression` field of the loop.
+
+For example, to iterate over a list provided as the Flow's input, 
+called `myList`, you would use `result.myList` as the iterator expression.
+
+For-loops add an extra object to the execution context, called `iter`, 
+which is nested under both `flow_input` and `previous_result`.
```

Comment:
For previous_result, it is only true for the first step of the for loop. A better way to see this is that a for loop has an embedded flow and the embedded flow has a separate flow_input made of the iterator as you said, the flow_input of the embedding flow.
It can also has input transforms to map arbitrary values from the embedding flow but we do not expose that yet.

Reply:
Added extra explanation in an admonition, reworded things a bit.